app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
```

Comment:
- [ ] This method should be in the `post.rb` model as mentioned in the project requirements. Please, move it there
![image](https://github.com/ichsansandy/gemstone-blog/assets/45460192/a674f2d9-fa22-4987-827e-32fc61099d7f)

[OPTIONAL]
- You can use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private for example:
```ruby
  after_save :update_post_counter

  private

  def update_posts_counter
    author.increment!(:posts_counter)
  end
```